TerrariaServerAPI/TerrariaApi.Server/ServerApi.cs

```diff
@@ -412,18 +415,46 @@ internal static void UnloadPlugins()
 			}
 		}
 
+		// Dictionary<full name, file path>
+		private static Dictionary<string, string> assemblyMap;
+
+		private static void PopulateAssemblyMap(string dir)
+		{
+			if (assemblyMap != null) return;
+
+			var paths = Directory.EnumerateFiles(dir, "*.dll", SearchOption.AllDirectories)
+				.Concat(Directory.EnumerateFiles(dir, "*.dll-plugin", SearchOption.AllDirectories))
+				.Distinct(StringComparer.OrdinalIgnoreCase);
+
+			assemblyMap = new Dictionary<string, string>();
+
+			foreach (var path in paths)
```

Comment:
I'm not a syntax nut but we should really get some braces in here. This Python style indentation based code hurts my head.